app/src/test/java/org/simple/clinic/user/UserSessionTest.kt

```diff
@@ -722,4 +732,85 @@ class UserSessionTest {
         listOf(Single.error<LoginResponse>(IOException()), false)
     )
   }
+
+  @Test
+  fun `user approved for syncing changes should be notified correctly`() {
+    fun createUser(loggedInStatus: User.LoggedInStatus, userStatus: UserStatus): List<User> {
+      return listOf(PatientMocker.loggedInUser(status = userStatus, loggedInStatus = loggedInStatus))
+    }
+
+    val userSubject = PublishSubject.create<List<User>>()
+    whenever(userDao.user())
+        .thenReturn(userSubject.toFlowable(BackpressureStrategy.BUFFER))
+
+    val observer = userSession.canSyncData().test()
+
+    userSubject.apply {
+      onNext(createUser(loggedInStatus = LOGGED_IN, userStatus = WAITING_FOR_APPROVAL))
```

Comment:
Keeping the emissions and the assertions makes this test a bit difficult to follow, especially considering that we have a lot of test values here. What do you think about clubbing them together?

Something along the lines of,

```kotlin
onNext(createUser(loggedInStatus = LOGGED_IN, userStatus = WAITING_FOR_APPROVAL))
observer.assertLatestValue(false)

onNext(createUser(loggedInStatus = LOGGED_IN, userStatus = APPROVED_FOR_SYNCING))
observer.assertLatestValue(true)
```

`TestObserver#assertLatestValue()` could be a local extension function:

```kotlin
TestObserver.assertLatestValue(value: T) {
  assertValueAt(observer.valueCount() - 1, value)
}
```

Reply:
This looks good. I'll try this out.